sktime/transformers/summarise.py

```diff
@@ -215,3 +228,286 @@ def transform(self, X, y=None):
+        -------
+        Xt : pandas DataFrame
+          Transformed pandas DataFrame
+        """
+        #input checks
+        validate_X(X)
+        # check_X_is_univariate(X)
+        X_time_series = convert_data(X)
+       
+
+        # check for default_fc_parameters
+        # TODO Is this method required? Drop handling this error to tsfresh?
+        self._check_default_rc_parameters()
+        
+        # TODO Checks for kind_to_fc_params
+
+        Xt = extract_features(
+                    X_time_series,
+                    column_id="index", column_value="value", 
+                    column_kind="column", column_sort="time_index",
+                    default_fc_parameters=self.passed_default_fc_params,
+                    kind_to_fc_parameters=self.passed_kind_to_fc_params,
+                    n_jobs=self.n_jobs, show_warnings=self.show_warnings,
+                    disable_progressbar=self.disable_progressbar,
+                    impute_function=self.impute_function,
+                    profile=self.profile,
+                    profiling_filename=self.profiling_filename,
+                    profiling_sorting=self.profiling_sorting,
+                    distributor=self.distributor)
+
+        return Xt
+
+
+    def transform_with_input(self,X,y=None):
+        #input checks
+        validate_X(X)
+
+        # set timeseries_container
+        if self.timeseries_container is None:
+            raise RuntimeError("You have to provide a time series using the set_timeseries_container function before.")
+
+        # check_X_is_univariate(X)
+        timeseries_container_converted = convert_data(self.timeseries_container)
+        
+        self._check_default_rc_parameters()
+        extracted_features = extract_features(timeseries_container_converted,
+                                                column_id="index", column_value="value", 
+                                                column_kind="column", column_sort="time_index",
+                                                default_fc_parameters=self.passed_default_fc_params,
+                                                kind_to_fc_parameters=self.passed_kind_to_fc_params,
+                                                n_jobs=self.n_jobs, show_warnings=self.show_warnings,
+                                                disable_progressbar=self.disable_progressbar,
+                                                impute_function=self.impute_function,
+                                                profile=self.profile,
+                                                profiling_filename=self.profiling_filename,
+                                                profiling_sorting=self.profiling_sorting,
+                                                distributor=self.distributor)
+
+        X_merged = pd.merge(X, extracted_features, left_index=True, right_index=True, how="left")
+
+        return X_merged
+
+
+
+class TsFreshFeatureSelector(BaseTransformer):
```

Comment:
Is this a feature selector or a relevant feature extractor (extraction and selection)?

Reply:
So this is similar to the RelevantFeatureExtractor
It transforms and fits in the fit function and then in the transform only selects the selected features using the `feature_selector.relevant_features` attribute.

Reply:
I'd just rename it to "RelevantFeatureExractor"

Reply:
Okay, will do